qutebrowser/browser/browsertab.py

```diff
@@ -806,6 +808,10 @@ def __init__(self, mode_manager: modeman.ModeManager,
         self._tab = tab
         self._mode_manager = mode_manager
 
+    @abstractmethod
```

Comment:
From the [`@abstractmethod` docs](https://docs.python.org/3/library/abc.html#abc.abstractmethod):

> Using this decorator requires that the class’s metaclass is `ABCMeta` or is derived from it.

We don't use `abc` here, for two reasons:

- Custom metaclasses like this usually don't play well with PyQt classes and inheriting from them.
- We don't need the runtime validation. If we have a normal method raising `NotImplementedError`, the `pylint` linter we use already makes sure it does get overridden.